src/embedded-checkout/embedded-checkout.spec.ts

```diff
@@ -149,4 +171,50 @@ describe('EmbeddedCheckout', () => {
         expect(messageListener.addListener)
             .toHaveBeenCalledWith(EmbeddedCheckoutEventType.FrameLoaded, options.onFrameLoad);
     });
+
+    it('configures styles when iframe is loaded', async () => {
+        options = { ...options, styles };
+        embeddedCheckout = new EmbeddedCheckout(
+            iframeCreator,
+            messageListener,
+            messagePoster,
+            loadingIndicator,
+            options
+        );
+
+        jest.spyOn(messagePoster, 'post');
+
+        await embeddedCheckout.attach();
+
+        expect(messagePoster.post).toHaveBeenCalledWith({
+            type: EmbeddedContentEventType.StyleConfigured,
+            payload: styles,
+        });
+    });
+
+    it('reconfigures styles when iframe is reloaded again', async () => {
+        options = { ...options, styles };
+        embeddedCheckout = new EmbeddedCheckout(
+            iframeCreator,
+            messageListener,
+            messagePoster,
+            loadingIndicator,
+            options
+        );
+
+        jest.spyOn(messagePoster, 'post');
+
+        await embeddedCheckout.attach();
+
+        messageListener.trigger({ type: EmbeddedCheckoutEventType.FrameLoaded });
+
+        expect(messagePoster.post).toHaveBeenCalledTimes(2);
+    });
+
+    it('toggles loading indicator', async () => {
+        await embeddedCheckout.attach();
+
+        expect(loadingIndicator.show).toHaveBeenCalled();
```

Comment:
can you do something like, it toggles loading indicator after checkout has loaded?

```
expect(loadingIndicator.show).toHaveBeenCalled();
expect(loadingIndicator.hide).not.toHaveBeenCalled();

frameLoaded()

expect(loadingIndicator.hide).toHaveBeenCalled();
```

you could also check it hides if frame failed to load.